dependencies.py

```diff
@@ -31,21 +31,20 @@ def extra_requirements() -> typing.Dict[str, typing.List[str]]:
     #     - We have a copy of these in extras-requirements.txt. If you modify these, make sure to change it
     #       there as well
     extras_require = {
-        # from 1.17.107 boto3 requires botocore>=1.20.107,<1.21.0 which
-        # conflicts with s3fs 2021.8.1 that has aiobotocore~=1.4.0
-        # which so far (1.4.1) has botocore>=1.20.106,<1.20.107
-        # boto3 1.17.106 has botocore>=1.20.106,<1.21.0, so we must add botocore explicitly
+        # last version that supports python 3.7: fsspec: 2023.1.0, aiobotocore: 2.4.2
+        # selecting fsspec 2023.1.0 and same version for the s3fs and gcsfs implementations
+        # s3fs 2023.1.0 requires aiobotocore 2.4.2 which requires botocore 1.27.59
+        # requesting boto3 1.24.59, the only version that requires botocore 1.27.59
         "s3": [
-            "boto3~=1.9, <1.17.107",
-            "botocore>=1.20.106,<1.20.107",
-            "aiobotocore~=1.4.0",
-            "s3fs~=2021.8.1",
+            "boto3~=1.24.59",
+            "aiobotocore~=2.4.2",
+            "s3fs~=2023.1.0",
         ],
         "azure-blob-storage": [
             "msrest~=0.6.21",
             "azure-core~=1.24",
             "azure-storage-blob~=12.13",
-            "adlfs~=2021.8.1",
+            "adlfs~=2022.2.0",
```

Comment:
Shouldn't this also be `2023.1.0`, as per fsspec and the other file system implementations?

Reply:
somehow missed it has an 2023.1.0 version, fixed accordingly

Reply:
actually (after the test fail) I now remember why adlfs was not at 2023.1.0 - it's not python-3.7 compatible. latest that does support python-3.7 is 2022.2.0
so we can push back all fsspec stack, but since 2022.2.0 dependency is fsspec>=2021.10.1 (and stay the same with latest adlfs) I figure we can just pin this package to 2022.2.0 and keep the rest of the fsspec at 2023.1.0